### ISSUE TITLE
Nested a tags do not throw an error

### ISSUE THREAD

**Issue by ChillerDragon** [linthtml/linthtml#809]
```
$ linthtml --version
0.10.1
$ cat foo.html 
<a href="#">
    <a href="#"></a>
</a>
$ linthtml foo.html 
✔ Found 1 files
✔ Files analyzed

File: foo.html
Using preset: default
 -:-  warning  Rule "id-class-no-ad" is deprecated.                                                       
 -:-  warning  Rule "id-class-style" is deprecated. Use the rules "id-style" and "class-style" instead.   

✖ 2 problems (0 error, 2 warnings)
$ 
```

Nested a tags are not supported by html. You can see chromium actually unnesting them in the devtools

![Image](https://github.com/user-attachments/assets/fc320957-bc61-4a21-9e46-4c10528e2d50)

**Comment by KamiKillertO** [comment]
This is because the HTML parser LintHTML uses does not fail when parsing the following.
What I can do is add a new rule `no-nested-interactive` that will report an error for this and also for things like `<a><button></button</a>`

**Comment by ChillerDragon** [comment]
That would be nice
